src/Nethermind/Nethermind.Optimism/Rpc/IOptimismEthRpcModule.cs

```diff
@@ -0,0 +1,337 @@
+// SPDX-FileCopyrightText: 2024 Demerzel Solutions Limited
+// SPDX-License-Identifier: LGPL-3.0-only
+
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using Nethermind.Blockchain.Find;
+using Nethermind.Core;
+using Nethermind.Core.Crypto;
+using Nethermind.Facade.Eth;
+using Nethermind.Facade.Filters;
+using Nethermind.Int256;
+using Nethermind.JsonRpc;
+using Nethermind.JsonRpc.Data;
+using Nethermind.JsonRpc.Modules;
+using Nethermind.JsonRpc.Modules.Eth;
+using Nethermind.State.Proofs;
+
+namespace Nethermind.Optimism;
+
+[RpcModule(ModuleType.Eth)]
+public interface IOptimismEthRpcModule : IRpcModule
```

Comment:
can this extend IEthRpcModule?

Reply:
Let me try to solve it via generics

Reply:
why would they be needed? You can have a return type of base class and then return derived type - it should serialize correctly in json rpc.

Reply:
We can use IResultWrapper, but we will lose type information. Which may be useful for tests, documentation generation and less convenient for devs.